_posts/2024-05-28-cockpit-317.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Cockpit 317
+author: jelle
+date: '2024-05-28'
+tags: cockpit, machines, files
+slug: cockpit-317
+category: release
+summary: 'webserver system user changes, Grafana setup now prefers Valkey instead of Redis'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly.
+
+Here are the release notes from Cockpit 317:
+
+
+## webserver: System user changes
+
+Cockpit has changed how the system user is handled. All supported distributions use the same system user names. We don't test, support, or recommend running the web server as root, which had previously been the default with an upstream build and install. Hence, the following `./configure` options related to static users have been removed:
+ - `--with-cockpit-user`
+ - `--with-cockpit-group`
+ - `--with-cockpit-ws-instance-user`
+ - `--with-cockpit-ws-instance-group`
+
+The `cockpit-ws` system user is no longer statically created, but created transiently and on-demand via [systemd's `DynamicUser` feature](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#DynamicUser=). If you would like to remove the previously-used static system user:
+ 1. Upgrade to Cockpit 317 or later
+ 2. Run `systemctl stop cockpit`
+ 3. Run `userdel -r cockpit-ws`
+
+The `cockpit-wsinstance` system user is now declared through [systemd-sysusers](https://www.freedesktop.org/software/systemd/man/latest/sysusers.d.html#).
+
+## Metrics: Grafana setup now prefers Valkey over Redis
+
+The [Grafana metrics setup](https://cockpit-project.org/blog/pcp-grafana.html) ("Export to network") now prefers using [valkey](https://valkey.io/) over [redis](https://redis.io/) on Fedora. See the [Fedora change proposal](https://fedoraproject.org/wiki/Changes/Replace_Redis_With_Valkey) for details.
```

Comment:
The Grafana metrics setup ("Export to network") now prefers... scans weird.

It would work fine without the ("Export to network") or if it was ("Expect to network" panel) or so.

Reply:
I dropped the "Export to network", that makes it a bit shorter.